boards/arm/nrf52_pca20020/board.c

```diff
@@ -53,7 +53,7 @@ static const struct pwr_ctrl_cfg vdd_pwr_ctrl_cfg = {
 	.pin = VDD_PWR_CTRL_GPIO_PIN,
 };
 
-DEVICE_INIT(vdd_pwr_ctrl_init, "", pwr_ctrl_init, NULL, &vdd_pwr_ctrl_cfg,
+DEVICE_INIT(NULL, vdd_pwr_ctrl_init, "", pwr_ctrl_init, NULL, &vdd_pwr_ctrl_cfg,
```

Comment:
I have a basic question: Why not define a new macro and leave the old macro redefined to call the new macro with this extra NULL? That way you don't break the current usage of the macro.

Reply:
In order to have working really device power management, we have to have full information about device dependencies. By changing existing macro I tried to "suggest" changes in existing drivers as well as "enforce" providing parent information in newly created ones.